src/Generator/EnumGenerator/CasesInterface.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Laminas\Code\Generator\EnumGenerator;
+
+interface CasesInterface
```

Comment:
This looks like a sealed type, where `CasesInterface = PureCases|BackedCases`

I suggest getting rid of the interface, and declaring the two types directly: this prevents implementing `CasesInterface` in userland, which would simplify our assumptions quite a bit.

Reply:
nevermind, doing it with DocBlocks is also good.